test/parallel/test-fs-write-file-invalid-path.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+
+const isWindows = process.platform === 'win32';
+
+if (!isWindows) {
+  common.skip('This test is for Windows only.');
+  return;
+}
+
+// refer https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx
+// ignore '/', '\\' and ':'
+const REVERSED_CHARACTERS = '<>"|?*';
+
+REVERSED_CHARACTERS.split('').forEach((ch) => {
+  assert.throws(
+    () => {
+      fs.writeFileSync(`c:\\somefile_${ch}_`, 'hello');
+    },
+    new RegExp(`no such file or directory, open 'c:\\\\somefile_\\${ch}_'`),
+    `failed with '${ch}'`);
+});
+
+// test for ':'
+// refer https://msdn.microsoft.com/en-us/library/windows/desktop/bb540537.aspx
+fs.writeFileSync('c:\\foo:bar', 'hello');
```

Comment:
Instead of `c:\\` use `common.tmpDir`
(Then somewhere near the top of the file run `common.refreshTmpDir();`)